rep-2010.rst

```diff
@@ -231,47 +257,77 @@ These topics are vulnerable to undesired actors listening in on service communic
+A user may want to avoid the overhead of publishing service event messages during normal operation,
+so it is desirable to be able to enable service introspection features during runtime.
+There is existing tooling to work with parameters on a per-node basis such as the ``parameter_client`` API in ``rclcpp`` or ``rclpy`` as well as for groups of nodes in ``ros2 launch`` or all nodes using wildcards in ``YAML`` parameter files [10]_.
+This makes ROS parameters the ideal choice as opposed to using environment variables or some other custom implementation.
+
+
+Default Services
+----------------
+
+ROS 2 parameters are built on top of ROS 2 service/clients and already publish out to ``/parameter_events`` on parameter changes [11]_.
+As parameters can be set directly by the host as well as through the corresponding parameter service, the service introspection proposed by this REP would fail to capture a subset of parameter events.
+For this reason service introspection will be disabled by default for ROS 2 parameters in favour of ``/parameter_events``. 
+This behavior can be changed through `a parameter <Configuration_>`_ if desired.
+
+
 Only supporting one service per name
 ------------------------------------
+It is technically possible to create more than one service on a given name (though not recommended).
+This use case will not be supported since enabling service introspection in such a case would result in ambiguous behaviour.
 
-TODO
 
 Separate request and response events instead of single service event
 --------------------------------------------------------------------
 
-TODO
+Separate events for requests and responses will be published instead of a single service event or 
+publishing an unique event for clients and services (for a total of 4 event types).
+This is to best balance avoiding introducing additional topic surface area and complexity of consuming emitted introspection data. 
+Grouping events by request or response rather than by client or subscriber would also ease request/response tracing work.
+
+.. Concern: our enable flags are enable by serv/client but our messages are grouped via req/response. Could be confusing?
+
 
 Optional content
 ----------------
 
-TODO
+Duplicating request or response content will be optional as service response messages can be *very* large and flooding the ROS network with up to four copies of the same message can introduce undesirable load and latency.
+By default content will be included on ``CLIENT_SENT`` and ``SERVICE_SENT``. 
+
 
 Backwards Compatibility
 =======================
 
-TODO
+Service introspection is a new feature which is not planned to impact any existing logic. 
+It is also an opt-in feature so there will be no additional incurred overhead by default.
+As always, code written to take advantage of this new feature will not be backwards compatible (at least regarding this feature) with versions of ROS 2 that do not have it implemented.
```

Comment:
I think this is a given and not worth mentioning.

Reply:
Minor refactor of this section: [308f7c5](https://github.com/jacobperron/rep/pull/2/commits/308f7c58622f97be29d7fab0f0831c4735058531)